### PR TITLE
Add `swipe_actions` modifier

### LINKED ISSUE #549: Modifier -> SwiftUI -> Lists: swipeActions(edge:allowsFullSwipe:content:)
Doc: https://developer.apple.com/documentation/swiftui/view/swipeactions(edge:allowsfullswipe:content:)

- [ ] Swift implementation
- [ ] Elixir implementation

### PR DESCRIPTION
**Pull request by supernintendo**
This PR adds support for the [swipeActions](https://developer.apple.com/documentation/swiftui/view/swipeactions(edge:allowsfullswipe:content:)) List modifier.

```heex
<VStack id="swipe_actions">
  <Text>
    <%= @swipe_event_message %>
  </Text>
  <Spacer />
  <List>
    <%= for {item, is_unread?} <- [{"Foo", true}, {"Bar", false}, {"Baz", false}] do %>
      <HStack
        id={item}
        modifiers={
          @native
          |> swipe_actions(edge: :leading, allows_full_swipe: false, content: :hello)
          |> swipe_actions(edge: :trailing, allows_full_swipe: false, content: :delete)
          |> swipe_actions(edge: :trailing, allows_full_swipe: false, content: :flag)
        }>
        <Text><%= item %></Text>
        <swipe_actions:hello>
          <%= if is_unread? do %>
            <Button>
              <Image system-name="envelope.badge" />
            </Button>
          <% else %>
            <Button modifiers={tint(@native, color: :accent)}>
              <Image system-name="envelope.open" />
            </Button>
          <% end %>
        </swipe_actions:hello>
        <swipe_actions:flag>
          <% tapping the flag_item event will update the @swipe_event_message assign %>
          <Button phx-click="flag_item" phx-value-item={item}>
            <Image system-name="flag" />
          </Button>
        </swipe_actions:flag>
        <swipe_actions:delete>
          <Button modifiers={tint(@native, color: :red)}>
            <Image system-name="trash" />
          </Button>
        </swipe_actions:delete>
      </HStack>
    <% end %>
  </List>
</VStack>
```

https://user-images.githubusercontent.com/5893007/235537826-ab584c8d-90f4-4fa3-b6e3-34b17eed2c5c.mov

Closes #549

### CODE REVIEW

**Review by shadowfacts — APPROVED** [review]

File: Sources/LiveViewNative/Modifiers/Lists Modifiers/SwipeActionsModifier.swift [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+///         <Button phx-click="delete_item" phx-value-item-id={@item.id} modifiers={tint(@native, color: :red)}>
+///           <Image system-name="trash" />
+///         </Button>
+///       </swipe_actions:delete>
+///     </HStack>
+///   <% end %>
+/// </List>
+///
+/// ## Arguments
+/// * ``allowsFullSwipe``
```
Comment (dginsburg) [inline Sources/LiveViewNative/Modifiers/Lists Modifiers/SwipeActionsModifier.swift:40]: The Elixir code will still use snake case: `allows_full_swipe`

File: Sources/LiveViewNative/Modifiers/Lists Modifiers/SwipeActionsModifier.swift [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    }
+
+    func body(content: Content) -> some View {
+        content.swipeActions(edge: edge, allowsFullSwipe: allowsFullSwipe) {
+            context.buildChildren(of: element, withTagName: self.content, namespace: "swipe_actions")
+        }
+    }
+
+    enum CodingKeys: String, CodingKey {
+        case allowsFullSwipe = "allows_full_swipe"
```
[the remark this answers is not in the document]
Comment (supernintendo) [reply Sources/LiveViewNative/Modifiers/Lists Modifiers/SwipeActionsModifier.swift]: Oh cool, thanks 🙂

**Review by shadowfacts — APPROVED** [review]

File: Sources/LiveViewNative/Modifiers/Lists Modifiers/SwipeActionsModifier.swift [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        default:
+            self.edge = .trailing
+        }
+        self.allowsFullSwipe = try container.decode(Bool.self, forKey: .allowsFullSwipe)
+        self.content = try container.decode(String.self, forKey: .content)
+    }
+
+    func body(content: Content) -> some View {
+        content.swipeActions(edge: edge, allowsFullSwipe: allowsFullSwipe) {
+            context.buildChildren(of: element, withTagName: self.content, namespace: "swipe_actions")
```
Comment (carson-katri) [inline Sources/LiveViewNative/Modifiers/Lists Modifiers/SwipeActionsModifier.swift:80]: This will need to be updated since I merged #787

**Review by carson-katri — CHANGES_REQUESTED** [review]

File: Sources/LiveViewNative/Modifiers/Lists Modifiers/SwipeActionsModifier.swift [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    #if swift(>=5.8)
+    @_documentation(visibility: public)
+    #endif
+    /// A boolean value that indicates whether the swipe action should be allowed to span the entire width of the screen.
+    let allowsFullSwipe: Bool
+    #if swift(>=5.8)
+    @_documentation(visibility: public)
+    #endif
+    /// The content of the swipe action.
+    let content: String
```
Comment (carson-katri) [inline Sources/LiveViewNative/Modifiers/Lists Modifiers/SwipeActionsModifier.swift:59]: The `@_documentation` attribute should be on the property, not the doc comment.
```suggestion
    /// The edge of the list item to associate the swipe actions with.
    #if swift(>=5.8)
    @_documentation(visibility: public)
    #endif
    let edge: HorizontalEdge
    /// A boolean value that indicates whether the swipe action should be allowed to span the entire width of the screen.
    #if swift(>=5.8)
    @_documentation(visibility: public)
    #endif
    let allowsFullSwipe: Bool
    /// The content of the swipe action.
    #if swift(>=5.8)
    @_documentation(visibility: public)
    #endif
    let content: String
```

**Review by carson-katri — CHANGES_REQUESTED** [review]

**Review by carson-katri — APPROVED** [review]

**Pull request merged**